src/services/quotes/quote-sources/sovryn-source.ts

```diff
@@ -0,0 +1,74 @@
+import { Chains } from '@chains';
+import { Chain, Address } from '@types';
+import { Addresses } from '@shared/constants';
+import { IQuoteSource, QuoteParams, QuoteSourceMetadata, QuoteSourceSupport, SourceQuoteRequest, SourceQuoteResponse } from './types';
+import { addQuoteSlippage, failed } from './utils';
+import { isSameAddress } from '@shared/utils';
+import { DEFAULT_SWAP_ROUTES, SmartRouter, smartRoutes } from '@sovryn/sdk';
+import { AlwaysValidConfigAndContextSource } from './base/always-valid-source';
+
+export const SOVRYN_METADATA: QuoteSourceMetadata<SovrynSupport> = {
+  name: 'Sovryn',
+  supports: {
+    chains: [Chains.ROOTSTOCK.chainId],
+    swapAndTransfer: false,
+    buyOrders: false,
+  },
+  logoURI: 'ipfs://QmUpdb1zxtB2kUSjR1Qs1QMFPsSeZNkL21fMzGUfdjkXQA',
+};
+
+type SovrynSupport = { buyOrders: false; swapAndTransfer: false };
+export class SovrynQuoteSource extends AlwaysValidConfigAndContextSource<SovrynSupport> {
+  getMetadata() {
+    return SOVRYN_METADATA;
+  }
+
+  async quote({
+    components: { providerService },
+    request: {
+      chain,
+      sellToken,
+      buyToken,
+      order,
+      accounts: { takeFrom },
+      config: { slippagePercentage },
+    },
+  }: QuoteParams<SovrynSupport>): Promise<SourceQuoteResponse> {
+    // Get Rootstock provider
+    const provider = providerService.getEthersProvider({ chainId: Chains.ROOTSTOCK.chainId });
+    // Create smart router
+    // FIX: Enable routes ZeroRedemption, MoC and Mynt once they enable it's use without permit.
+    const smartRouter = new SmartRouter(provider, Object.values([smartRoutes.ammSwapRoute]));
+    // Find best quote
+    try {
+      const result = await smartRouter.getBestQuote(mapNativeToken(sellToken), mapNativeToken(buyToken), order.sellAmount);
+      // Get swap tx data
+      const swapTxData = await result.route.swap(mapNativeToken(sellToken), mapNativeToken(buyToken), order.sellAmount, takeFrom, {
+        slippage: 500,
+      });
```

Comment:
@0xsambugs I'm not sure how to handle this slippage + ours. Is there any example about this? I could do 99% slippage here, and just use ours.

Reply:
Just set it like this: `slippage: slippagePercentage * 100`,